tidy3d/components/data/dataset.py

```diff
@@ -257,6 +258,90 @@ class FieldDataset(ElectromagneticFieldDataset):
         description="Spatial distribution of the z-component of the magnetic field.",
     )
 
+    def from_zbf(filename: str, dim1: xyz, dim2: xyz):
+        """Creates a :class:`.FieldDataset` from a Zemax Beam File (zbf)
+
+        Parameters
+        ----------
+        filename: str
+            Specification of the source time-dependence.
+        dim1: xyz
+            Tangential field component to map the x-axis of the zbf data to.
+            eg. `dim1 = "z"` sets `FieldDataset.Ez` to Ex of the zbf data.
+        dim2: xyz
+            Tangential field component to map the y-axis of the zbf data to.
+
+        Returns
+        -------
+        :class:`.FieldDataset`
+            A :class:`.FieldDataset` object with two tangential E field components populated
+            by zbf data.
+        """
+        log.warning(
+            "`FieldDataset.from_zbf()` is currently an experimental feature."
```

Comment:
it's confusing but for console outputs you will want the code to be in regular single quotes like '

Reply:
this comes up in the ValueErrors and things below, so just double check all of them 👍

Reply:
https://www.notion.so/flexcompute/Python-Style-Guide-afac5dbbf949463e89c18fe39b7453c9